Alerters/slack.py

```diff
@@ -29,25 +29,28 @@ def __init__(self, config_options):
         else:
             channel = None
 
+        if 'username' in config_options:
+            username = config_options['username']
+        else:
+            username = None
+
         if url == "":
             raise RuntimeError("missing url")
 
         self.url = url
         self.channel = channel
+        self.username = username
 
     def send_alert(self, name, monitor):
         """Send the message."""
 
-        if self.channel is None:
-            return
-
         type = self.should_alert(monitor)
         (days, hours, minutes, seconds) = self.get_downtime(monitor)
 
         host = "on host %s" % self.hostname
 
-        if self.channel is not None:
```

Comment:
This deletion leaves the value of self.channel unused. Should this maybe be something like,
```python
if self.channel is not None:
    ...
elif self.username is not None:
    ...
else:
    ...
```
instead?

Reply:
Yes, omitting channel would be a problem.